diff --git a/build/src/Drivers/S3.js b/build/src/Drivers/S3.js
index 074ac24b7acc185d73a86af6ff7331a33c197c96..f2ba5547ddb6641d4db716f81cbbadb6770fb9b3 100644
--- a/build/src/Drivers/S3.js
+++ b/build/src/Drivers/S3.js
@@ -217,7 +217,7 @@ class S3Driver {
         if (this.config.cdnUrl) {
             return `${this.config.cdnUrl}/${location}`;
         }
-        const href = (0, url_1.format)(await this.adapter.config.endpoint());
+        const href = (0, url_1.format)((await this.adapter.config?.endpoint?.()) || '');
         if (href.startsWith('https://s3.amazonaws')) {
             return `https://${this.config.bucket}.s3.amazonaws.com/${location}`;
         }